L46-core-tls-credential-API.md

```diff
@@ -0,0 +1,339 @@
+Title
+----
+* Author(s): ZhenLian
+* Approver: markdroth, yashykt, jiangtaoli2016
+* Status: Draft
+* Implemented in:
+* Last updated: September 7, 2020
+
+## Abstract
+
+This proposal aims to enhance the security features provided by [this already implemented proposal](https://github.com/grpc/proposal/pull/98). 
+The improvements include:
+1. replacing the per-connection callback-based credential reloading mechanism with the new distributor/provider mechanism
+2. changing the server authorization check to an authorization check that could be applied to both the client side and the server side
+3. adding options for users to configure their TLS version 
+
+## Background
+
+The current [TLS credential API](https://github.com/grpc/grpc/blob/master/src/core/lib/security/credentials/tls/tls_credentials.h) in gRPC C core implemented the following two features:
```

Comment:
This link points to internal implementation details, not to the C-core API.  While it does make sense for this proposal to discuss implementation details that are relevant to understanding the C-core API, we should focus more on the latter and include only those implementation details that are important for understanding the API.